Blog post translation (Error Handling in React 16) (#116)
reactjs/fr.reactjs.org: content/blog/2017-07-26-error-handling-in-react-16.md

---

> title: "Gestion des erreurs dans React 16"

Reviewer: ```suggestion
title: "La gestion des erreurs dans React 16"
```

En VF un article est souvent plus naturel

author: [gaearon]
---

> Comme la version de React 16 se rapproche, nous souhaitons annoncer quelques modifications de la façon dont React traite les erreurs JavaScript dans les composants. Ces modifications sont incluses dans les versions bêta de React 16 et feront partie de React 16.

Reviewer: ```suggestion
Alors que la version de React 16 se rapproche, nous aimerions annoncer quelques modifications dans la façon dont React traite les erreurs JavaScript au sein des composants. Ces modifications figurent dans les versions beta de React 16 et feront partie de React 16.
```

Outre la fluidité, « bêta » en français est plus une légère insulte qu'autre chose, on évite en général le circonflexe dans le cadre informatique.

> **D'ailleurs, [nous venons de publier la première version bêta de React 16 pour que vous puissiez l'essayer !](https://github.com/facebook/react/issues/10294)**

Reviewer: ```suggestion
**D'ailleurs, [nous venons de publier la première version beta de React 16 pour que vous puissiez l'essayer !](https://github.com/facebook/react/issues/10294)**
```

> ## Comportement dans React 15 et antérieur {#behavior-in-react-15-and-earlier}

Reviewer: ```suggestion
## Comportement jusqu’à React 15 inclus {#behavior-in-react-15-and-earlier}
```

}
```

Mais les composants React sont déclaratifs et spécifient *ce qui* doit être rendu :

```js
<Button />
```

Les périmètres d'erreurs respectent la nature déclarative de React, et se comportent sans surprises. Par exemple, même si une erreur survient dans une méthode `componentDidUpdate` suite à un `setState` quelque part au fin fond de l'arbre des composants, elle se propagera correctement jusqu’au périmètre d'erreur le plus proche.

## Changements de nommage par rapport à React 15 {#naming-changes-from-react-15}

> React 15 disposait d'une prise en charge très limitée des périmètres d'erreurs sous un nom de méthode différent : `unstable_handleError`. Cette méthode ne fonctionne plus, et vous devrez la remplacer par `componentDidCatch` dans votre code à partir de la première version bêta de React 16.

Reviewer: ```suggestion
React 15 disposait d'une prise en charge très limitée des périmètres d'erreurs sous un nom de méthode différent : `unstable_handleError`. Cette méthode ne fonctionne plus, et vous devrez la remplacer par `componentDidCatch` dans votre code à partir de la première version beta de React 16.
```

> Pour ce changement, nous fournissons [un codemod](https://github.com/reactjs/react-codemod#error-boundaries) qui vous permet de migrer automatiquement votre code

Reviewer: ```suggestion
Pour ce changement, nous fournissons [un codemod](https://github.com/reactjs/react-codemod#error-boundaries) qui vous permet de migrer automatiquement votre code.
```
